src/child-applications/lifecycles/unload.js

```diff
@@ -13,23 +13,34 @@ export async function toUnloadPromise(app) {
 		 * anything that still thinks we need to unload the app.
 		 */
 		finishUnloadingApp(app, unloadInfo);
-		return;
+		return app;
+	}
+
+	if (app.status === UNLOADING) {
+		/* Both unloadChildApplication and reroute want to unload this app.
+		 * It only needs to be done once, though.
+		 */
+		await unloadInfo.promise;
+		return app;
 	}
 
-	if (app.status !== NOT_MOUNTED || !unloadInfo) {
-		/* Either no one has called unloadChildApplication for this app,
-		 * or it's not in a status where it can be unloaded.
+	if (app.status !== NOT_MOUNTED) {
```

Comment:
This is the same as before, just split the if statement out into two if statements